hastexo/hastexo.py

```diff
@@ -343,14 +340,6 @@ def parse_xml(cls, node, runtime, keys, id_generator):
 
         return block
 
-    def add_dict_properties_to_node(self, properties, node):
-        """
-        Add properties from a dictionary as attributes to `node`.
-
-        """
-        for key in properties.keys():
-            node.set(key, str(properties.get(key, '')))
```

Comment:
Just a question: I guess another way to do this would be to just keep using this method, albeit with

```python
for key in sorted(properties.keys()):
```

... which would sort the keys asciibetically. But I'm sure you thought of that, so I'm curious what advantages you found out the explicit open-coded approach has. :)

Reply:
Mainly to handle a missing key with either throwing a KeyError (port name & number, provider name) or setting a default (True for hook_events & -1 for capacity). It ended up looking more readable like this for me but I can give it another look.

Reply:
No, that's fine. More readable is good. :)